### PR TITLE
[apps] move ble handlers to common server

### PR DESCRIPTION
**Pull request by gjc13**
#### Problem

Network provision over BLE prototype is missing in the light app.

 #### Summary of Changes

Move BLE network provision code from lock app to the common server.

### CODE REVIEW

File: examples/common/chip-app-server/Server.cpp [hunk 1]
```diff
@@ -47,13 +53,75 @@ using namespace ::chip::DeviceLayer;
...
+    ChipLogProgress(AppServer, "BLE Connection closed");
+}
+
+void HandleBLEMessageReceived(chip::Ble::BLEEndPoint * endPoint, chip::System::PacketBuffer * buffer)
+{
+#if CHIP_ENABLE_OPENTHREAD
+    uint16_t bufferLen = buffer->DataLength();
+    uint8_t * data     = buffer->Start();
+    chip::DeviceLayer::Internal::DeviceNetworkInfo networkInfo;
+    ChipLogProgress(AppServer, "Receive message size %u", bufferLen);
```
Comment (bukepo) [inline examples/common/chip-app-server/Server.cpp:77]: How about also log the peer's BLE address?

```suggestion
    ChipLogProgress(AppServer, "Receive BLE message. size=%u peer=%s", bufferLen, XXX);
```

File: examples/common/chip-app-server/Server.cpp [hunk 2]
```diff
@@ -47,13 +53,75 @@ using namespace ::chip::DeviceLayer;
...
     return sessions;
 }
+
 } // namespace chip
 
 namespace {
+
+void HandleBLEConnectionClosed(chip::Ble::BLEEndPoint * endPoint, BLE_ERROR err)
+{
+    ChipLogProgress(AppServer, "BLE Connection closed");
```
Comment (bukepo) [inline examples/common/chip-app-server/Server.cpp:68]: Log error as well?

**Review by bukepo — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by vivien-apple — APPROVED** [review]

**Comment by andy31415** [comment]
@woody-apple ?

File: examples/common/chip-app-server/Server.cpp [hunk 3]
```diff
@@ -47,13 +53,79 @@ using namespace ::chip::DeviceLayer;
...
+
+void HandleBLEMessageReceived(chip::Ble::BLEEndPoint * endPoint, chip::System::PacketBuffer * buffer)
+{
+#if CHIP_ENABLE_OPENTHREAD
+    uint16_t bufferLen = buffer->DataLength();
+    uint8_t * data     = buffer->Start();
+    chip::DeviceLayer::Internal::DeviceNetworkInfo networkInfo;
+    ChipLogProgress(AppServer, "Receive BLE message size=%u", bufferLen);
+
+    memcpy(networkInfo.ThreadNetworkName, data, sizeof(networkInfo.ThreadNetworkName));
```
Comment (bzbarsky-apple) [inline examples/common/chip-app-server/Server.cpp:83]: This is moving existing code, but shouldn't this have error checking to make sure we have enough data in the buffer for all of this stuff we're doing and that we don't read off the end of the buffer?  Please make sure there's a TODO comment here and a followup issue filed.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/common/chip-app-server/Server.cpp (hunk 3 above)
Comment (gjc13) [reply examples/common/chip-app-server/Server.cpp:85]: Buffer length check added.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**